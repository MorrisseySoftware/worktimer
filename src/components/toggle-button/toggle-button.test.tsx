import React from 'react'
import { act, create, ReactTestRenderer } from 'react-test-renderer'
import { shallow, mount, render } from 'enzyme'
import ToggleButton from './toggle-button'

describe('Toggle Button Tests', () => {
    it('should toggle the button state from OFF to ON', () => {
        let component: ReactTestRenderer = {} as ReactTestRenderer
        act(() => {
            component = create(
                <ToggleButton off={true} clickHandler={() => {}}></ToggleButton>
            )
        })
        expect(component.toJSON()).toMatchSnapshot()
        act(() => {
            component.update(
                <ToggleButton
                    off={false}
                    clickHandler={() => {}}
                ></ToggleButton>
            )
        })
        expect(component.toJSON()).toMatchSnapshot()
    })

    it('should call a click handler when the toggle button is clicked', () => {
        var testState = ''
        var testHandler = () => {
            testState = 'called!'
        }
        const component = shallow(
            <ToggleButton off={true} clickHandler={testHandler}></ToggleButton>
        )
        component.find('button').simulate('click')
        expect(testState).toBe('called!')
    })
})
