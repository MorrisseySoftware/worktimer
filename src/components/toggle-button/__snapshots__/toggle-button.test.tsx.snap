// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Button Tests should toggle the button state from OFF to ON 1`] = `
<button
  className="toggle__btn button--off"
  onClick={[Function]}
>
  START TIMER
</button>
`;

exports[`Toggle Button Tests should toggle the button state from OFF to ON 2`] = `
<button
  className="toggle__btn button--on"
  onClick={[Function]}
>
  STOP TIMER
</button>
`;
